labelbox/schema/data_row.py

```diff
@@ -17,6 +19,35 @@
 logger = logging.getLogger(__name__)
 
 
+class KeyType(str, Enum):
+    ID = 'ID'
+    """An existing CUID"""
+    GKEY = 'GKEY'
+    """A Global key, could be existing or non-existing"""
+    AUTO = 'AUTO'
+    """The key will be auto-generated. Only usable for creates"""
+
+
+class ResolvableId(BaseModel):
+    """
+    The ResolvableId class is a unique ID abstraction that allows us to reference
+    a DataRow by either a Global key or CUID
+    """
+    type: KeyType = KeyType.GKEY
+    value: str
+
+
+class DataRowSpec(BaseModel):
```

Comment:
This will ultimately need to support metadata, attachments, embeddings, .etc, essentially all the fields supported by import.